core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+
+        collection_rights_model = collection_rights_model_tuples[0][1]
+        version = item.id[item.id.rfind('-') + 1:]
+        if int(collection_rights_model.version) < int(version):
+            cls.errors['collection rights model version check'] = (
+                'Model id %s: Collection Rights model corresponding to '
+                'id %s has a version %s which is less '
+                'than the version %s in snapshot content model id' % (
+                    item.id, collection_rights_model.id,
+                    collection_rights_model.version, version))
+
+    @classmethod
+    def _get_validation_functions(cls):
+        return [cls._validate_collection_model_version_from_item_id]
+
+
+class CollectionCommitLogEntryModelValidator(BaseModelValidator):
+    """Class for validating CollectionCommitLogEntryModel."""
+
+    is_commit_log_entry_model = True
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        regex_string = '(collection|rights)-%s-\\d*$' % (
+            item.collection_id)
+
+        return regex_string
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return commit_commands_domain.CollectionCommitCmd
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'collection_model': (
+                collection_models.CollectionModel, [item.collection_id]),
+        }
+
+    @classmethod
+    def _validate_collection_model_version_from_item_id(cls, item):
+        """Validate that collection model corresponding to item.collection_id
+        has a version greater than or equal to the exp version in item.id.
+
+        Args:
+            item: CollectionCommitLogEntryModel to validate.
+        """
+        _, collection_model_tuples = cls.external_models['collection_model']
+
+        collection_model = collection_model_tuples[0][1]
+        version = item.id[item.id.rfind('-') + 1:]
+        if int(collection_model.version) < int(version):
+            cls.errors['collection model version check'] = (
+                'Model id %s: Collection model corresponding to collection '
+                'id %s has a version %s which is less than the version %s in '
+                'commit log model id' % (
+                    item.id, item.collection_id, collection_model.version,
+                    version))
+
+    @classmethod
+    def _get_validation_functions(cls):
+        return [cls._validate_collection_model_version_from_item_id]
+
+
+class CollectionSummaryModelValidator(BaseModelValidator):
+    """Class for validating CollectionSummaryModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        non_negative_int_schema = {
```

Comment:
Should we also be adding a schema for `contributors_summary` field?

Reply:
I don't think we can do that since the keys in dict are user_ids and so we cannot define a specific schema for keys and how many of them exist.